lib/elixir/lib/task.ex

```diff
@@ -341,10 +339,12 @@ defmodule Task do
   Receives a group of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct. If the message matches a
+  task that terminated abnormally, this function will exit.
```

Comment:
Yes, it was meant to mirror await. Plus we cannot change it because it
would be backwards incompatible.

On Wednesday, December 30, 2015, Alexei Sholik <notifications@github.com>
wrote:

> In lib/elixir/lib/task.ex
> <https://github.com/elixir-lang/elixir/pull/4128#discussion_r48607917>:
>
> > @@ -341,10 +339,12 @@ defmodule Task do
> >    Receives a group of tasks and a message and finds
> >    a task that matches the given message.
> >
> > -  This function returns a tuple with the returned value
> > -  in case the message matches a task that exited with
> > -  success alongside the matching task. It returns `nil`
> > -  if no task was found. It exits if the task has failed.
> > +  If the message matches a task that finished normally,
> > +  this function returns a tuple with the task's
> > +  result and `Task` struct. If the message matches a
> > +  task that terminated abnormally, this function will exit.
>
> OK, so I had to read the source code to finally understand what this
> function is doing. What do you think about the following wording:
>
>   If the message matches a task that finished normally,
>   this function returns a tuple with the task's
>   result and `Task` struct. If the message matches a
>   task that terminated before sending its result back,
>   this function will exit.
>
> I didn't want to say just "terminated" because that is more likely to
> raise question about the exact semantics.
>
> As a slight offtopic, the exiting behaviour of this function surprises me.
> If the common use case for find is to check whether an incoming message
> is related to a task, I would expect to get an error tuple back for a
> prematurely terminated task. I guess this current behaviour is supposed to
> mirror what happens when you await?
>
> /cc @josevalim <https://github.com/josevalim>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4128/files#r48607917>.
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*